doc/source/core_developer.md

```diff
@@ -135,6 +135,12 @@ speak up in case they are not fine with the current state of the pull request.
 Another exceptional situation is one where the first approving review happened
 a long time ago and many changes have taken place in the meantime.
 
+When squashing the commits, please edit the commit messages if necessary, to
+ensure they are standard and they will read nicely in the commit history.
+Typically reorganize them, deleting or condensing the less informative ones
+such as "pep8 fix", "apply review comments", etc. Please make sure to maintain
+all Co-authored-by entries.
+
```

Comment:
Sorry, @mkcor, I was half asleep yesterday and didn't catch that this paragraph also includes the use of "standard".

```suggestion
When squashing commits Github concatenates all commit messages.  Please edit the resulting message so that it gives a concise, tidy overview of changes.  For example, you may want to grab the description from the PR itself, and delete lines such as "pep8 fix", "apply review comments", etc. Please retain all Co-authored-by entries.
```

Reply:
No worries, it's not too late! :smile: 
I think that if someone is a core developer, they probably know of 'standard' commit messages, so this case is a little different, I would argue. That being said, I like the overall rewording you suggested for this paragraph. About the form:
- I prefer writing 'GitHub' instead of 'Github' (as can be found in the rest of the file);
- I prefer wrapping lines at 79 characters (as done in the rest of the file).

Reply:
:+1: That is the correct capitalization of GitHub, thanks!  And line wrapping is a mess when making suggestions, let me feed it through my editor.  OK, done.

Reply:
Thank you, @stefanv! Syncing up locally, I caught these leftovers: be56fcf726f4ae3a39385489b481aab5b4056a93 (iiuc, it takes two spaces after a period in plain English, but not in Markdown because it's up to the renderer to... render the proper spacing; like LaTeX, right).

Reply:
It's an old habit, originating from the time of typewriters, that was taught to me when I was young: back then, the editors/"word processors" didn't space things out automatically, so you added two spaces to create a slightly bigger space between sentences. It isn't necessary anywhere any more; but as they say: old habits die hard.